src/main/webapp/js/instructorFeedbackEdit/customFeedbackPaths.js

```diff
@@ -0,0 +1,358 @@
+var FEEDBACK_PARTICIPANT_TYPE_STUDENTS = 'STUDENTS';
+var FEEDBACK_PARTICIPANT_TYPE_INSTRUCTORS = 'INSTRUCTORS';
+var FEEDBACK_PARTICIPANT_TYPE_TEAMS = 'TEAMS';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM = 'OWN_TEAM';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS = 'OWN_TEAM_MEMBERS';
+var FEEDBACK_PARTICIPANT_TYPE_OWN_TEAM_MEMBERS_INCLUDING_SELF = 'OWN_TEAM_MEMBERS_INCLUDING_SELF';
+var FEEDBACK_PARTICIPANT_TYPE_NONE = 'NONE';
+var TEAM_NAME_INSTRUCTORS = 'Instructors';
+
+var sessionCreator;
+var studentEmailToTeamNameMap;
+var instructorEmails;
+var teamNameToStudentEmailsMap;
+var studentEmails;
+var teamNames;
+var allPossibleFeedbackGivers;
+var allPossibleFeedbackRecipients;
+
+$(document).ready(function() {
+    initialiseCustomFeedbackPathsData();
+    initialiseFeedbackPathsSpreadsheets();
+    bindEventHandlers();
+});
+
+function initialiseCustomFeedbackPathsData() {
+    sessionCreator = $('#session-creator-data').data('session-creator');
+    studentEmailToTeamNameMap = $('#students-data').data('students');
+    instructorEmails = $('#instructors-data').data('instructors');
+    teamNameToStudentEmailsMap = [[]];
+    studentEmails = [];
+    teamNames = [];
+    
+    for (var studentEmail in studentEmailToTeamNameMap) {
+        if (studentEmailToTeamNameMap.hasOwnProperty(studentEmail)) {
+            studentEmails.push(studentEmail);
+            
+            var teamName = studentEmailToTeamNameMap[studentEmail];
+            if (!teamNames.includes(teamName)) {
+                teamNames.push(teamName);
+            }
+            
+            var studentEmailsList = teamNameToStudentEmailsMap[teamName];
+            if (studentEmailsList === undefined) {
+                studentEmailsList = [];
+            }
+            studentEmailsList.push(studentEmail);
+            teamNameToStudentEmailsMap[teamName] = studentEmailsList;
+        }
+    }
+    
+    allPossibleFeedbackGivers = [''];
+    allPossibleFeedbackGivers = allPossibleFeedbackGivers.concat(studentEmails);
+    for (var i = 0; i < instructorEmails.length; i++) {
+        if (!allPossibleFeedbackGivers.includes(instructorEmails[i])) {
+            allPossibleFeedbackGivers.push(instructorEmails[i]);
+        }
+    }
+    allPossibleFeedbackGivers = allPossibleFeedbackGivers.concat(teamNames);
+    allPossibleFeedbackRecipients = allPossibleFeedbackGivers.slice();
+    allPossibleFeedbackRecipients.push(TEAM_NAME_INSTRUCTORS);
+    allPossibleFeedbackRecipients.push('%GENERAL%');
+}
+
+function initialiseFeedbackPathsSpreadsheets() {
+    $('.form_question').each(function() {
+        var $questionForm = $(this);
+        var giverType = $questionForm.find('select[id^="' + FEEDBACK_QUESTION_GIVERTYPE + '"]').val();
+        var recipientType = $questionForm.find('select[id^="' + FEEDBACK_QUESTION_RECIPIENTTYPE + '"]').val();
+        generateFeedbackPathsSpreadsheet($questionForm.find('.custom-feedback-paths-spreadsheet'), giverType, recipientType);
```

Comment:
I passed in the `$questionForm.find('.custom-feedback-paths-spreadsheet')` object as it is the actual `$container` object that is used to init the handsontable. I thought this would be more appropriate for the `generateFeedbackPathsSpreadsheet` method as it specifically means "generating the feedback paths spreadsheet/handsontable" to this `$container`

Although, at this point I'm not sure which approach would be better, I might find that one way may be more suited for my purposes as development of this feature progresses.